### ISSUE TITLE
allow @language x without modes/x.py (when x is provided by pygments)

### ISSUE THREAD

**Issue by tangentstorm** [leo-editor/leo-editor#2484]
See PR  #2486 and PR #2491.

Pygments provides lexers for languages that aren't present in `modes/*.py`, but `BaseColorizer.init_mode` explicitly rejects languages that don't have a corresponding file in the leo/modes directory:
```python
    # Don't try to import a non-existent language.
    path = g.os_path_join(g.app.loadDir, '..', 'modes')
    fn = g.os_path_join(path, f"{language}.py")
    if g.os_path_exists(fn):
        mode = g.import_module(name=f"leo.modes.{language}")
    else:
        mode = None
```
After enabling `@bool use-pygments = True`, I managed to get leo to recognize the language I wanted ("k") by explicitly creating a completely empty `k.py` in the leo install directory. (K isn't part of the latest pygments release, but it is in the pygments repo under [q.py]( https://github.com/pygments/pygments/blob/20c9141b1589f1baea5e209a523f7985cf89fb77/pygments/lexers/q.py))

Anyway, it would be nice if leo would accept language commands for languages that are only defined in pygments.

**Comment by edreamleo** [comment]
@tangentstorm Excellent idea!  Why didn't I think to do that? I'll do this issue next.

**Comment by tbpassin** [comment]
If @language is going to accept these other languages, remember to make sure that their values are reflected in the "kind" attribute for the node.  VR3 will thank you!

**Comment by edreamleo** [comment]
@tangentstorm The ekr-pygments branch (PR #2486) contains an experimental fix for this issue. Please test this branch and report your results.  Thanks.

**Comment by tbpassin** [comment]
In VR, and mostly borrowed in VR3:

`       def get_kind(self, p):
    """Return the proper rendering kind for node p."""
`
That's what I mean.  I think it's basically what _@language_ is in effect for the page.

I was thinking that the "kind" was returned by higher Leo machinery.  Now that I have looked it up, I see that it's coming from within VR/VR3 themselves, and not something to worry about  for implementing this enhancement.  Might need to do something in VR3 at some point.

**Comment by edreamleo** [comment]
@tbpassin Thanks for the clarification.  I'll merge this branch into devel soon.

**Comment by tangentstorm** [comment]
Working great for me on the `devel` branch. Thank you so much! :)

**Comment by tangentstorm** [comment]
After using it a bit, I've noticed two things:

- the `@language whatever` now gets highlighted even if neither pygments or leo provides the language. Previously, it would fail to highlight the language name if it were unrecognized.
- (more importantly) child nodes don't seem to inherit the language the way they normally do

**Comment by edreamleo** [comment]
@tangentstorm Thanks for the continued testing. I'll fix this issue for 6.6.

**Comment by edreamleo** [comment]
#2491 contains my latest attempt. The actual "real" change was small. However, I also refactored two three colorizer classes.  See the PR for details.
